internal/docs/rest/rest.go

```diff
@@ -0,0 +1,252 @@
+--------
+
+.. code-block::
+`
+
+// GenCustom creates custom reStructured Text output.
+// Adapted from github.com/spf13/cobra/doc to match MongoDB tooling and style
+func GenCustom(cmd *cobra.Command, w io.Writer, linkHandler func(string, string) string) error {
+	cmd.InitDefaultHelpCmd()
+	cmd.InitDefaultHelpFlag()
+
+	buf := new(bytes.Buffer)
+	name := cmd.CommandPath()
+
+	short := cmd.Short
+	long := cmd.Long
+	if long == "" {
+		long = short
+	}
+	ref := strings.ReplaceAll(name, " ", "_")
+
+	buf.WriteString(".. _" + ref + ":\n\n")
+	buf.WriteString(strings.Repeat("=", len(name)) + "\n")
+	buf.WriteString(name + "\n")
+	buf.WriteString(strings.Repeat("=", len(name)) + "\n")
+	buf.WriteString(toc)
+	buf.WriteString("\n" + short + "\n")
+	buf.WriteString("\n" + long + "\n\n")
+
+	if cmd.Runnable() {
+		buf.WriteString(syntaxHeader)
+		buf.WriteString(fmt.Sprintf("\n   %s\n\n", strings.ReplaceAll(cmd.UseLine(), "[flags]", "[options]")))
+	}
+	printArgsReST(buf, cmd)
+	printOptionsReST(buf, cmd)
+
+	if len(cmd.Example) > 0 {
+		buf.WriteString(examplesHeader)
+		buf.WriteString(fmt.Sprintf("\n%s\n\n", indentString(cmd.Example, " ")))
+	}
+
+	if hasSeeAlso(cmd) {
+		buf.WriteString("See Also\n")
+		buf.WriteString("--------\n\n")
+		if cmd.HasParent() {
+			parent := cmd.Parent()
+			pname := parent.CommandPath()
+			ref = strings.ReplaceAll(pname, " ", "_")
+			buf.WriteString(fmt.Sprintf("* %s \t - %s\n", linkHandler(pname, ref), parent.Short))
+			cmd.VisitParents(func(c *cobra.Command) {
+				if c.DisableAutoGenTag {
+					cmd.DisableAutoGenTag = c.DisableAutoGenTag
+				}
+			})
+		}
+
+		children := cmd.Commands()
+		sort.Sort(byName(children))
+
+		for _, child := range children {
+			if !child.IsAvailableCommand() || child.IsAdditionalHelpTopicCommand() {
+				continue
+			}
+			cname := name + " " + child.Name()
+			ref = strings.ReplaceAll(cname, " ", "-")
+			buf.WriteString(fmt.Sprintf("* %s \t - %s\n", linkHandler(cname, ref), child.Short))
+		}
+		buf.WriteString("\n")
+	}
+	if !cmd.DisableAutoGenTag {
+		buf.WriteString("*Auto generated by MongoDB CLI on " + time.Now().Format("2-Jan-2006") + "*\n")
+	}
+	_, err := buf.WriteTo(w)
+	return err
+}
+
+// Test to see if we have a reason to print See Also information in docs
+// Basically this is a test for a parent command or a subcommand which is
+// both not deprecated and not the autogenerated help command.
+func hasSeeAlso(cmd *cobra.Command) bool {
+	if cmd.HasParent() {
+		return true
+	}
+	for _, c := range cmd.Commands() {
+		if !c.IsAvailableCommand() || c.IsAdditionalHelpTopicCommand() {
+			continue
+		}
+		return true
+	}
+	return false
+}
+
+type byName []*cobra.Command
+
+func (s byName) Len() int           { return len(s) }
+func (s byName) Swap(i, j int)      { s[i], s[j] = s[j], s[i] }
+func (s byName) Less(i, j int) bool { return s[i].Name() < s[j].Name() }
+
+const optionsHeader = `.. list-table::
+   :header-rows: 1
+   :widths: 20 10 10 60
+
+   * - Option
```

Comment:
If we're going to reuse this for all three tables, we'll need to think of a more generic word than "option", since that won't work for the "argument" section. Possibly "Field"? or just "Name"?

Reply:
I'm reusing because I'm lazy if you need them to be different feel free to suggest so